src/hooks/useConnectSid.ts

```diff
@@ -0,0 +1,97 @@
+import { useExec } from "@raycast/utils";
+import React from "react";
+import * as crypto from "crypto";
+
+function resolveConnectSid(connectSidHex: string, browserKeyHex: string) {
+  const encryptedValueWithoutHeader = Buffer.from(connectSidHex, "hex").slice(3);
+
+  const pass = Buffer.from(browserKeyHex, "hex");
+
+  const salt = Buffer.from("saltysalt");
+  const iv = Buffer.alloc(16, " "); // An empty initialization vector
+  const keyLength = 16;
+
+  const iterations = 1003;
+  const key = crypto.pbkdf2Sync(pass, salt, iterations, keyLength, "sha1");
+
+  const cipher = crypto.createDecipheriv("aes-128-cbc", key, iv);
+
+  const decrypted = cipher.update(encryptedValueWithoutHeader);
+  return decrypted.toString("ascii") + cipher.final("ascii");
+}
+
+function useBrowserKeyHex() {
+  const res = useExec(`security find-generic-password -s "Arc Safe Storage" -g -w`, { shell: true });
+
+  if (res.isLoading) {
+    return {
+      isLoading: true,
+    };
+  }
+
+  if (res.error) {
+    return {
+      isLoading: false,
+      error: res.error,
+    };
+  }
+
+  const browserKey = res.data;
+
+  if (!browserKey) {
+    throw new Error("No browser key found");
+  }
+
+  return {
+    isLoading: false,
+    browserKeyHex: Buffer.from(browserKey).toString("hex"),
+  };
+}
+
+function useConnectSidHex() {
+  const res = useExec(
+    `sqlite3 ~/Library/Application\\ Support/Arc/User\\ Data/Default/Cookies "select hex(encrypted_value) from cookies where host_key='replit.com' and name='connect.sid'"`,
```

Comment:
this has some issues, but should be easy to fix:

- as the path suggests, it works only for Arc, but it should be trivial to make it work for any other chromium browser
- we need to add logic to scan all your profiles for the token

Reply:
Can we store the response with raycast's storage API? Kind of annoying we have to re-approve every time the command is opened right now. you could also try `keepPreviousData: true`

Reply:
yeah for sure

Reply:
i just clicked "allow forever", but makes sense to cache it. 

the browser key is dangerous though, you can use it to access every secret on the browser. this isn't a replacement for a PAT / OAuth based API, but def the fastest way to ship an extension without adding a whole new auth system to replit